### ISSUE TITLE
Azure Monitor Datasource: Fetch metrics by Resource ID

### ISSUE THREAD

**Issue by jkroepke** [grafana/grafana#78372]
**Why is this needed**:

I have a datasource variables which contains a list of Azure Resource ID.

While I can metrics multiple metrics for multiple resources via Prometheus Datasource, I not able to do such an request for Azure Monitor Datasource.

**What would you like to be added**:

I would like to fetch multiple Azure Monitor metrics by Resource ID 

**Who is this feature for?**

In Grafana, I'm using a table to display all VMs in my Tenant (via ResourceGraph) and I would like to enrich the table with metrics from Azure Monitor

**Comment by usmangt** [comment]
Hi @jkroepke,

Thanks for opening this issue.

Maybe this community post has the answer?

https://community.grafana.com/t/integration-with-azure-metrics-variables-not-getting-full-scope/43216

**Comment by jkroepke** [comment]
Does not seems like this. The mention posted tries to resolve something different.

**Comment by adamyeats** [comment]
@jkroepke Would you be able to screenshot (or copy and paste) the query you are trying to run against Azure Monitor, and add it here please?

If I am understanding correctly, whereas [this post linked above](https://community.grafana.com/t/integration-with-azure-metrics-variables-not-getting-full-scope/43216) gets a list of all the available resource groups, you are instead looking to get all the metrics from a list of resource IDs that you already have. Is that correct?

**Comment by jkroepke** [comment]
@adamyeats 

> you are instead looking to get all the metrics from a list of resource IDs that you already have. Is that correct?

Not 100% correct. I'm looking NOT looking for ALL metrics, I'm looking to get one specific metric from a list of resource ID.

My use-case is an Dashboard that uses a Table and a Variables. 

The variable `resourceID` is a multi-value variables. It contains the resourceID of Virtual Machines

The Table Panel has multiple query from different datasources.

![image](https://github.com/grafana/grafana/assets/1560587/b31c8bbb-e4bc-446f-9002-1462b6761f3a)

The first query asks data from Azure Resource Graph via Azure Grafana Datasource.

```
resources
| where  $__contains(id, $resourceID)
```

The second query asks Prometheus 

```
avg(100 - ((node_memory_MemAvailable_bytes{resourceID=~"$resourceID"} / node_memory_MemTotal_bytes) * 100)) by (resourceID)
```

Now I want to do the same for Azure Monitor. 

For example, I would like to get the VM Availability Metric from Azure Monitor for all resources in `$resourceID`.


> Would you be able to screenshot (or copy and paste) the query you are trying to run against Azure Monitor

Basically it not possible the do this in one Rest Call. A individual request is required for each Resource.

```
for $resourceId in $resourceIds; do
  https://management.azure.com/$resourceId/providers/microsoft.Insights/metrics?interval=PT5M&metricnames=Available Memory Bytes&aggregation=average&metricNamespace=microsoft.compute%2Fvirtualmachines&autoadjusttimegrain=true&validatedimensions=false&api-version=2019-07-01
done
```

The Azure Portal is able to batch multiple requests efficiently, but it seems like the GoSDK is not able to use the Batch API.

![Bildschirmfoto 2023-12-05 um 15 25 22](https://github.com/grafana/grafana/assets/1560587/af847dde-4485-423e-939a-4b1911c0ae06)

**Comment by adamyeats** [comment]
@jkroepke Thanks for clarifying! 🚀 Unfortunately, this isn’t supported at the moment, the Azure Monitor plugin does not currently support querying metrics by resource URI from the frontend. I'll go ahead and convert this issue to a feature request, and we'll add this to our backlog for future planning, and we'll be sure to update this issue if/when this gets picked up.

**Comment by jkroepke** [comment]
> Unfortunately, this isn’t supported at the moment

Thanks for the clarification. I'm aware that this is not supported and that the root cause of the issue. :)

**Comment by jkroepke** [comment]
Still up
